nexus/src/app/image.rs

```diff
@@ -24,54 +25,77 @@ use omicron_common::api::external::LookupResult;
 use omicron_common::api::external::LookupType;
 use omicron_common::api::external::NameOrId;
 use omicron_common::api::external::ResourceType;
+use omicron_common::api::external::UpdateResult;
 use std::str::FromStr;
 use std::sync::Arc;
 use uuid::Uuid;
 
 impl super::Nexus {
-    pub fn image_lookup<'a>(
+    pub async fn image_lookup<'a>(
         &'a self,
         opctx: &'a OpContext,
         image_selector: params::ImageSelector,
-    ) -> LookupResult<lookup::Image<'a>> {
+    ) -> LookupResult<ImageLookup<'a>> {
```

Comment:
The return type here is an enum that wraps either a `ProjectImage` or a `SiloImage`. It's important that we returned the appropriate type for authz purposes.